pkg/db/postgres.go

```diff
@@ -53,13 +71,46 @@ func (pc *PostgresConnection) Query(query string, args ...interface{}) (DataRows
 
 func (pc *PostgresConnection) Exec(query string, args ...interface{}) (sql.Result, error) {
 	pc.logger.Debugf("Postgres Exec(): %s | %v", query, args)
+	if !pc.validate(query) {
+		pc.logger.Errorf("Regex validation failed for query '%s'", query)
+		if !pc.executeUnverified {
+			return nil, fmt.Errorf("Regex validation failed for query '%s'", query)
+		}
+	}
 	result, err := pc.db.Exec(query, args...)
 	if err != nil {
 		pc.logger.Errorf("Postgres Exec() error: %s", err)
 	}
 	return result, err
 }
 
+func (pc *PostgresConnection) validate(query string) bool {
```

Comment:
Please see me comment below.. maybe we can centlraize this logical part in a dedicated validator object to encapsulate this part and make it easier testable.